syntax = "proto2";

package wireless_android_play_analytics;

// Next tag: 3
message EventPredicate {
  // Next tag: 4
  message MessageMatcher {
    optional string input_field = 1;
    optional int64 int_comparator = 2;
  }

  // Next tag: 3
  message UiElementPredicate {
    // The type of UI element proto message, e.g. store UI element, movie UI
    // element, etc.
    optional UiElementType ui_element_type = 1;

    // The matching rule for the UI element. Multiple ui_element_matcher are
    // AND'ed.
    repeated MessageMatcher ui_element_matcher = 2;
  }

  // Next tag: 4
  enum UiElementType {
    UNKNOWN_UI_ELEMENT = 0;
    STORE_UI_ELEMENT = 1;
    MOVIE_UI_ELEMENT = 2;
    PHOTO_UI_ELEMENT = 3;
  }

  // Next tag: 4
  message IndexUiMatcher {
    optional UiElementType ui_element_type = 1;
    optional int32 index = 2;
    optional MessageMatcher ui_element_matcher = 3;
  }

  message UiElementPathPredicate {
    optional UiElementPredicate leaf_ui_predicate = 1;
    optional UiElementType ui_element_type = 2;
    repeated IndexUiMatcher index_ui_matcher = 3;
  }
  // Defines the rules to match GWSLogEntryProto. Multiple event_matcher are
  // AND'ed.
  repeated MessageMatcher event_matcher = 1;

  // Defines the rules to match UiElementPath. Multiple UiElementPath are
  // OR'ed.
  repeated UiElementPathPredicate ui_element_path_predicate = 2;
}
